### PR TITLE
implement react-query for all content views

### PR DESCRIPTION
**Pull request by Pilopa**
## Proposed changes

Implements react-query and replaces direct apiClient usage for sources, annotate and history content views.

### Your checklist for this pull request

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [ ] New code is covered with unit tests
- [ ] New ui components are tested inside the storybook (module ui-components only) 
- [ ] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

**Comment by Pilopa** [comment]
No changelog entry as we have not done any for previous react-query-related performance changes except the first

File: scm-ui/ui-api/src/history.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const queryClient = useQueryClient();
+  const link = (file._links.history as Link).href;
+
+  const queryParams: Record<string, string> = {};
+  if (request?.page) {
+    queryParams.page = request.page.toString();
+  }
+
+  return useQuery<ChangesetCollection, Error>(
+    repoQueryKey(repository, "history", file.revision, request?.page || 0),
```
Comment (eheimbuch) [inline scm-ui/ui-api/src/history.ts:50]: I wonder why you only use the `file.revision` but not the branch for the query key. E.g. for sources we use both. But the caching seems not to work for sources, so idk.

**Review by eheimbuch — COMMENTED** [review]
Nice how much code we got rid off!

File: scm-ui/ui-api/src/history.ts (hunk 1 above)
Comment (Pilopa) [reply scm-ui/ui-api/src/history.ts:50]: Yeah I gotta admit I was not overly confident with the query keys. Maybe it is better for them to be overly precise. I will adjust it here.

Comment (Pilopa) [reply scm-ui/ui-api/src/history.ts:50]: The loading spinner actually comes from the `getContent` function within the `SourcecodeViewer`. I added it to the list of edge cases that we need to sort out as well.

**Pull request merged**